### PR TITLE
Added the option to set the text and body on erroneus responses

### LINKED ISSUE #16: Response body on errors
I'm testing a simple api service, with a bit of help of http://reqr.es/

Here is my mock config:

```javascript
export default [{
  pattern: 'http://reqr.es(.*)',

  // Fixtures return data
  fixtures: function( match, params, headers ) {

    if( match[1] === '/api/login' ) {
      if( params['password'] ) {
        return { "token": "QpwL5tke4Pnpja7X" };
      }
      throw new Error( 400 )
    }
  },

  // Requests return responses
  post: function( match, data ) {
    return {
      code: 200,
      body: data
    };
  }
}]
```

Now if I throw error 400 (for making an invalid request to `/api/login`) I need to verify the error being returned (in this concrete case `{ "error": "Missing password" }`).

How can I do that?

### PR DESCRIPTION
**Pull request by tomazzaman**
Use like this:

```javascript
fixtures: function (match, params, headers) {
      var error = new Error( 422 );
      var code = (match || [])[1] || 422;
      var newErr = new Error(parseInt(code));
      newErr.response = http.STATUS_CODES[code];
      newErr.status = code;
      newErr.responseHeader = 'application/json';
      newErr.responseText = '{"password": "missing"}';
      newErr.responseBody = { password: 'missing' }; // only needed on server/isomorphic
      throw newErr;
},
```

Fix #16

### CODE REVIEW

File: lib/superagent-mock.js [hunk 1]
```diff
@@ -109,15 +109,18 @@ function mock (superagent, config) {
...
+            on: function (){},
+            body: err.responseBody || {}
           },
           req: {
             method: function (){}
           },
+
           xhr: {
             responseType: '',
+            responseText: err.responseText || '',
```
Comment (fdubost) [inline lib/superagent-mock.js]: What is the difference between `responseBody` and `responseText` ?

File: tests/support/config.js [hunk 2]
```diff
@@ -68,6 +68,23 @@ module.exports = [
     }
   },
   {
+    pattern: 'https://validation.example',
+    fixtures: function (match, params, headers) {
+      var error = new Error( 422 );
```
Comment (fdubost) [inline tests/support/config.js]: This line is useless, no ? :smiley_cat:

File: lib/superagent-mock.js (hunk 1 above)
Comment (tomazzaman) [inline lib/superagent-mock.js]: Client-side, `response.body` gets created by parsing `responseText` and `responseHeader` - see [here](https://github.com/visionmedia/superagent/blob/master/lib/client.js#L316) 
Server-side, `response.body` gets just assigned to whatever the body of the response already is - see [here](https://github.com/visionmedia/superagent/blob/master/lib/node/response.js#L38)

I needed to include both for all tests to pass :)

File: tests/support/config.js (hunk 2 above)
Comment (tomazzaman) [inline tests/support/config.js]: Yeah, most likely, but can't hurt having it around :)

File: lib/superagent-mock.js (hunk 1 above)
Comment (fdubost) [inline lib/superagent-mock.js]: Ok, thanks for this explanation :wink:

File: tests/support/config.js (hunk 2 above)
Comment (fdubost) [inline tests/support/config.js]: I'd rather remove the unused code :smile_cat:

Comment (fdubost) [inline tests/support/config.js]: Could you remove this line, so we can merge this PR @tomazzaman :wink:

**Comment by fdubost** [comment]
ping @tomazzaman :angel:

**Comment by BenJBailey** [comment]
+1 on getting this merged :D

**Pull request merged**